testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/x509/X509BrowserLoginSubjectDnTest.java

```diff
@@ -0,0 +1,129 @@
+
+/**
+ * @author Sebastian Loesch
+ * @date 02/14/2019
+ */
+
+public class X509BrowserLoginSubjectDnTest extends AbstractX509AuthenticationTest {
+
+    @Page
+    protected AppPage appPage;
+
+    @Page
+    protected X509IdentityConfirmationPage loginConfirmationPage;
+
+    @Page
+    protected LoginPage loginPage;
+
+    @BeforeClass
+    public static void onBeforeTestClass() {
+        if (Boolean.parseBoolean(System.getProperty("auth.server.jboss"))) {
+            String authServerHome = System.getProperty("auth.server.home");
+
+            if (authServerHome != null && System.getProperty("auth.server.ssl.required") != null) {
+                authServerHome = authServerHome + "/standalone/configuration";
+                StringBuilder cliArgs = new StringBuilder();
+
+                cliArgs.append("--ignore-ssl-errors=true ");
+                cliArgs.append("--web-security=false ");
+                cliArgs.append("--ssl-certificates-path=" + authServerHome + "/ca.crt ");
+                cliArgs.append("--ssl-client-certificate-file=" + authServerHome + "/certs/clients/test-user-san-email@localhost.cert.pem ");
+                cliArgs.append("--ssl-client-key-file=" + authServerHome + "/certs/clients/test-user@localhost.key.pem ");
+                cliArgs.append("--ssl-client-key-passphrase=password");
+
+                System.setProperty("keycloak.phantomjs.cli.args", cliArgs.toString());
+            }
+        }
+    }
+
+    private String setup(boolean canonicalDnEnabled) throws Exception {
+        String clientCertFilePath = "x509/test-user-san-email@localhost.cert.pem";
+        URL clientCertUrl = (getClass().getClassLoader().getResource(clientCertFilePath));
+        if (clientCertUrl == null) {
+            throw new IOException("File not found: " + clientCertFilePath);
+        }
+        CertificateFactory cf = CertificateFactory.getInstance("X.509", "SUN");
+        X509Certificate clientCert = (X509Certificate)cf.generateCertificate(new FileInputStream(clientCertUrl.getFile()));
+        String subjectDn = canonicalDnEnabled ?
+          clientCert.getSubjectX500Principal().getName(X500Principal.CANONICAL) :
+          clientCert.getSubjectDN().getName();
+
+        UserRepresentation user = findUser("test-user@localhost");
+        user.singleAttribute("x509_certificate_identity",subjectDn);
+        updateUser(user);
+        return subjectDn;
+    }
+
+    private void login(X509AuthenticatorConfigModel config, String userId, String username, String attemptedUsername) {
+        AuthenticatorConfigRepresentation cfg = newConfig("x509-browser-config", config.getConfig());
+        String cfgId = createConfig(browserExecution.getId(), cfg);
+        Assert.assertNotNull(cfgId);
+
+        loginConfirmationPage.open();
+
+        Assert.assertTrue(loginConfirmationPage.getSubjectDistinguishedNameText().equals("CN=test-user, OU=Keycloak, O=Red Hat, L=Boston, ST=MA, C=US"));
```

Comment:
Prefer
`assertThat(loginConfirmationPage.getSubjectDistinguishedNameText(), is("CN=test-user, OU=Keycloak, O=Red Hat, L=Boston, ST=MA, C=US"));`
This has much better output in case of failure

Reply:
I will fix that in X509BrowserLoginSubjectAltNameEmailTest.java too, alright?

Reply:
Absolutely, thank you